### PR TITLE
contrib/k8s: use provided image when installing external workload

### PR DESCRIPTION
**Pull request by tklauser**
When installing external workloads, launch Cilium agent from the
provided, versioned image instead of hardcoding cilium/cilium:latest.
The following command will now use Cilium 1.9.4 instead of latest:

    CLUSTER_ADDR=<ip> CILIUM_IMAGE=cilium/cilium:v1.9.4 ./install-external-workload.sh`

Fixes: a1fac6aa2b56 ("contrib/k8s: Add scripts for provisioning VMs")

### CODE REVIEW

**Review by aanm — APPROVED** [review]

**Comment by tklauser** [comment]
No need to run full CI, should be good to merge once approved.

**Pull request merged**
